GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -394,71 +333,84 @@ private void Update()
             _statusIsUpToDate = true;
             _previousUpdateTime = Environment.TickCount;
 
+            // capture a consistent state in the main thread
+            var statusCancellationToken = _statusCancellationToken;
+            IGitModule module = Module;
```

Comment:
I don't think this will save us, should a user change working directory while we doing this.
Should it be something like 
```
var workingDir = Module.WorkingDir;
ThreadHelper.JoinableTaskFactory.RunAsync(async () => { ....
    var module = new GitModule(workingDir);
    ...
});